Cargo.toml

```diff
@@ -71,6 +71,8 @@ members = [
     "lib/rust/parser/ast",
     "lib/rust/parser/flexer",
     "lib/rust/prelude",
+    "lib/rust/profiler",
+    "lib/rust/profiler-macros",
```

Comment:
We are normally putting macros and implementation in a common folder. Here that should be `lib/rust/profiler/macros` and `lib/rust/profiler/impl`. This way, such layout is more future-proof. For example, if we would like to extract that as a separate library to its dedicated repository in the future, the `lib/rust/profiler` would become the repository `src` folder. We have the same layout already in `lib/rust/parser/flexer-testing`, although the naming there is not the best, and I think that we should stay with `/macros` and `/impl` naming instead (I'm very open for other name proposals!).

Reply:
I'd prefer to _not_ make "impl" directory; instead, let the implementation be placed in lib/rust/profiler and macros be a "sub-crate" in lib/rust/profiler/macros. This is more intuitive when filling up the dependencies (I constantly forget putting "impl" in paths).